lib/modules/datasource/deb/readme.md

```diff
@@ -55,3 +64,57 @@ RUN apt-get update && \
 ```
 
 When the apt package for `gcc` is updated, Renovate updates the environment variable.
+
+```json title="Override deb registryUrl with a packageRules entry"
+{
+  "packageRules": [
+    {
+      "matchDatasources": ["deb"],
+      "matchPackageNames": ["gcc-11"],
+      "registryUrls": [
+        "https://deb.debian.org/debian?suite=stable&components=main,contrib,non-free&binaryArch=amd64"
+      ]
+    }
+  ]
+}
+```
+
+## Artifactory
+
+The Debian datasource can be used with Artifactory.
+
+### Supported repository types
+
+The `debian` datasource supports these repository types:
+
+- virtual
+- local
+- remote
+
+### Set a `registryUrl`
+
+To use Artifactory, first configure the `deb` datasource by setting the `registryUrl`.
+
+```title="Example of valid registryUrl format"
+https://<host>:<port>/artifactory/<repository-slug>?release=<release>&components=<components>&binaryArch=<binaryArch>
+https://artifactory.example.com:443/artifactory/debian?release=bookworm&components=main,contrib,non-free&binaryArch=amd64
```

Comment:
@secustor It should be noted somewhere that these are fake URLs (the URL parameters after `?` are not valid parameters for Debian/Artifactory URLs, they are only used to configure renovate).

This means they can not be tested for validity by entering them in the browser without removing the params first. Unlike, for example, the nuget URL, which works when entered in the browser.

This had me scratching my head for a few minutes until I read the renovate source code and saw that they are parsed out before actually accessing the URL.

Reply:
They should be ignored by the backends, though it makes sense to be explicit about that. 

I have added a note panel regarding for that.

Reply:
Thanks!